packages/wonder-blocks-core/components/unique-id-provider.js

```diff
@@ -0,0 +1,66 @@
+// @flow
+import * as React from "react";
+
+import NoSSR from "./no-ssr.js";
+
+import UniqueIDFactory from "../util/unique-id-factory.js";
+import SsrIDFactory from "../util/ssr-id-factory.js";
+
+import type {IIdentifierFactory} from "../util/types.js";
+
+type Props = {|
+    // A render prop that takes an instance of IIdentifierFactory and returns
+    // the content to be rendered.
+    // If mockOnFirstRender is false, this is only called after
+    // the initial render has occurred and will always be called with the same
+    // IIdentifierFactory instance.
+    // If mockOnFirstRender is true, this is called once with
+    // a mock IIdentifierFactory for the initial render, and then a unique ID
+    // factory thereafter.
+    children: (IIdentifierFactory) => React.Node,
```

Comment:
Unfortunately, styleguidist won't replace `IIdentifierFactory` with the full signature of this callback which makes this bad from a documentation perspective.  There's a task in Jira to do something about this, but we haven't gotten around to it yet. :(  In the meantime can you use the full signature here?

Reply:
Ugh, that's a maintenance nightmare as it decouples the type definitions from one another just to satisfy the docs. Is there not a better way? I think I'd rather hand type the interface definition into some part of the docs than weaked our code type checking for the sake of documentation.

Reply:
I hand typed the interface definition into some part of the docs rather than weaken our code type checking for the sake of documentation. It was not fun though; I raised an issue on why.

Reply:
Yeah, typing it in the docs sounds like a good interim solution.